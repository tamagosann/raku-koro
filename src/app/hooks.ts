import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from './store';

// useDispatchとかuseSelectorの代わりにこっちを使う
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
