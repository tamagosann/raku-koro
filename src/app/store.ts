import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import dailyInfectionRuducer from "../features/graphs/dailyInfectionSlice";
import prefectureRuducer from "../features/common/prefectureSlice";
import dailyDeadRuducer from "../features/graphs/dailyDeadSlice";
import totalCoronaRuducer from "../features/graphs/totalCoronaSlice";
import totalDethRuducer from "../features/graphs/totalDethSlice";
import vaccinationRuducer from "../features/graphs/vaccinationSlice";
import bedOccupancyRateRuducer from "../features/graphs/bedOccupancyRateSlice";
import dailyPositiveRuducer from "../features/graphs/dailyPositiveSlice";

export const store = configureStore({
  reducer: {
    dailyInfection: dailyInfectionRuducer,
    prefecture: prefectureRuducer,
    dailyDead: dailyDeadRuducer,
    totalCorona: totalCoronaRuducer,
    totalDeth: totalDethRuducer,
    vaccination: vaccinationRuducer,
    bedOccupancyRate: bedOccupancyRateRuducer,
    dailyPositive: dailyPositiveRuducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
