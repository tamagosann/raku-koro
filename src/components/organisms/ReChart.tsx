import React from 'react';
import {
  Legend,
  AreaChart,
  Area,
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Brush,
  ResponsiveContainer,
} from 'recharts';

// 型のインポート
import { Data } from '../../features/graphs/dailyInfectionSlice';

interface Props {
  targetPrefecture: Array<Data>;
  children: string;
  dataKey: string;
  startIndex: number;
  endIndex: number;
  value: string;
  date: string;
}

export const ReChart = ({
  targetPrefecture,
  children,
  dataKey,
  startIndex,
  endIndex,
  value,
  date,
}: Props) => {
  return (
    <>
      {!Number(value) ? (
        <ResponsiveContainer
          width="100%"
          minHeight={600}
          data-testid="area-chart-tag"
        >
          <AreaChart
            width={1200}
            height={600}
            data={targetPrefecture}
            margin={{ top: 50, right: 30, left: 20, bottom: 5 }}
          >
            {/* 第一引数が色のついている大きさ、第二引数が幅間 */}
            <CartesianGrid strokeDasharray="30 5" stroke="#eee"></CartesianGrid>
            <XAxis dataKey="date" tick={{ fontSize: '.7rem' }} />
            <YAxis unit="人" />
            <Tooltip />
            <Legend
              stroke="#fd7e00"
              verticalAlign="top"
              wrapperStyle={{ lineHeight: '40px' }}
              fill="#fd7e00"
            />
            <Brush
              className="TimeLineChart-brush"
              dataKey={date}
              stroke="#fd7e00"
              startIndex={startIndex}
              endIndex={endIndex}
            />
            <Area
              dataKey={dataKey}
              stroke="#fd7e00"
              name={children}
              strokeWidth={3}
              unit="人"
              fill="#fd7e00"
            />
          </AreaChart>
        </ResponsiveContainer>
      ) : (
        <ResponsiveContainer
          width="100%"
          minHeight={600}
          data-testid="line-chart-tag"
        >
          <LineChart
            width={1200}
            height={600}
            data={targetPrefecture}
            margin={{ top: 50, right: 30, left: 20, bottom: 5 }}
          >
            {/* 第一引数が色のついている大きさ、第二引数が幅間 */}
            <CartesianGrid strokeDasharray="30 5" stroke="#eee"></CartesianGrid>
            <XAxis dataKey="date" tick={{ fontSize: '.7rem' }} />
            <YAxis unit="人" />
            <Tooltip />
            <Legend
              stroke="#fd7e00"
              verticalAlign="top"
              wrapperStyle={{ lineHeight: '40px' }}
              fill="#fd7e00"
            />
            <Brush
              className="TimeLineChart-brush"
              dataKey={date}
              stroke="#fd7e00"
              startIndex={startIndex}
              endIndex={endIndex}
            />
            <Line
              name={children}
              dataKey={dataKey}
              stroke="#fd7e00"
              strokeWidth={3}
              unit="人"
              fill="#fd7e00"
            />
          </LineChart>
        </ResponsiveContainer>
      )}
    </>
  );
};
