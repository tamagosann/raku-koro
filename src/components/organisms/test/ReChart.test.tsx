import React from 'react';
import {
  render,
  cleanup,
  screen,
  fireEvent,
  waitFor,
  within,
  getByRole,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { afterEach, expect } from '@jest/globals';
import { store } from '../../../app/store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { ReChart } from '../ReChart';
import { PropTypes } from '@material-ui/core';
import { Variant } from '@material-ui/core/styles/createTypography';

describe('TypographyTitleコンポーネント', () => {
  interface Data {
    daily_infection: number;
    total_infection: number;
    daily_dead: number;
    total_dead: number;
    date: string;
    pref_code: number;
    pref_name: string;
  }

  // デモProps
  interface ReChartPropsInterface {
    targetPrefecture: Array<Data>;
    children: string;
    dataKey: string;
    startIndex: number;
    endIndex: number;
    value: string;
    date: string;
  }

  // // 各テストの直前に実行される（コンポーネントの取得）
  // beforeEach(() => {
  //   render(
  //     <Provider store={store}>
  //       <ReChart
  //         targetPrefecture={ReChartProps.targetPrefecture}
  //         children={ReChartProps.children}
  //         dataKey={ReChartProps.dataKey}
  //         startIndex={ReChartProps.startIndex}
  //         endIndex={ReChartProps.endIndex}
  //         value={ReChartProps.value}
  //         date={ReChartProps.date}
  //       />
  //     </Provider>
  //   );
  // });

  // 各処理が通った後にアンマウントさせる
  afterEach(() => cleanup());

  describe('jest renderer test', () => {
    test('render correctly', () => {
      const ReChartProps: ReChartPropsInterface = {
        targetPrefecture: [
          {
            daily_infection: 10,
            total_infection: 20,
            daily_dead: 15,
            total_dead: 25,
            date: '2021-08-15',
            pref_code: 1,
            pref_name: '北海道',
          },
        ],
        children: 'テスト',
        dataKey: 'total_dead',
        startIndex: 0,
        endIndex: 31,
        value: '1',
        date: '2021-07-05',
      };
      const tree = renderer
        .create(
          <Provider store={store}>
            <ReChart
              targetPrefecture={ReChartProps.targetPrefecture}
              children={ReChartProps.children}
              dataKey={ReChartProps.dataKey}
              startIndex={ReChartProps.startIndex}
              endIndex={ReChartProps.endIndex}
              value={ReChartProps.value}
              date={ReChartProps.date}
            />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('三項演算子の条件分岐テスト', () => {
    test('false（０）のケース', () => {
      const ReChartProps: ReChartPropsInterface = {
        targetPrefecture: [
          {
            daily_infection: 10,
            total_infection: 20,
            daily_dead: 15,
            total_dead: 25,
            date: '2021-08-15',
            pref_code: 1,
            pref_name: '北海道',
          },
        ],
        children: 'テスト',
        dataKey: 'total_dead',
        startIndex: 0,
        endIndex: 31,
        value: '0',
        date: '2021-07-05',
      };

      render(
        <Provider store={store}>
          <ReChart
            targetPrefecture={ReChartProps.targetPrefecture}
            children={ReChartProps.children}
            dataKey={ReChartProps.dataKey}
            startIndex={ReChartProps.startIndex}
            endIndex={ReChartProps.endIndex}
            value={ReChartProps.value}
            date={ReChartProps.date}
          />
        </Provider>
      );
    });
    test('true（1）のケース', () => {
      const ReChartProps: ReChartPropsInterface = {
        targetPrefecture: [
          {
            daily_infection: 10,
            total_infection: 20,
            daily_dead: 15,
            total_dead: 25,
            date: '2021-08-15',
            pref_code: 1,
            pref_name: '北海道',
          },
        ],
        children: 'テスト',
        dataKey: 'total_dead',
        startIndex: 0,
        endIndex: 31,
        value: '1',
        date: '2021-07-05',
      };

      render(
        <Provider store={store}>
          <ReChart
            targetPrefecture={ReChartProps.targetPrefecture}
            children={ReChartProps.children}
            dataKey={ReChartProps.dataKey}
            startIndex={ReChartProps.startIndex}
            endIndex={ReChartProps.endIndex}
            value={ReChartProps.value}
            date={ReChartProps.date}
          />
        </Provider>
      );
    });
  });
});
