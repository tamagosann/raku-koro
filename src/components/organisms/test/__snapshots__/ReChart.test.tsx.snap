// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypographyTitleコンポーネント jest renderer test render correctly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": "100%",
      "maxHeight": undefined,
      "minHeight": 600,
      "minWidth": undefined,
      "width": "100%",
    }
  }
/>
`;
